library/ssl_tls.c

```diff
@@ -7722,7 +7722,7 @@ static int ssl_calc_finished_tls_generic(mbedtls_ssl_context *ssl, void *ctx,
 
     MBEDTLS_SSL_DEBUG_BUF(3, "calc finished result", buf, len);
 
-    mbedtls_platform_zeroize(padbuf, sizeof(padbuf));
+    mbedtls_platform_zeroize(padbuf, hlen);
```

Comment:
Damn, this should have been caught by review when it was split into `ssl_calc_finished_tls_sha256()` and `ssl_calc_finished_tls_sha384()`

Maybe we should scan the code for calls to `mbedtls_platform_zeroize()` with `sizeof(foo)` where `foo` is a pointer type. Hmm, surely a compiler could do this for us...

Reply:
AFAIK Clang doesn't have this warning, but clang-tidy does. Setting up clang-tidy is one of many quality improvements I'd like to do, time permitting. I expect that the task of setting it up would mostly be figuring out annotations or whitelisting of existing false positives.

Reply:
Unfortunately there isn't an annotation that could be used, but there are "ways" round that: see #8143, which found a similar issue in tests